### PR TITLE
[ABM-Edge-4] Implement frontend-service and move common classes into an SDK

### PR DESCRIPTION
**Pull request by Shabirmean**
### Depends on:
- Please review _(or make sure **merged** status of)_ the following PR before starting on this: 
   - #3 

### What's this?
- Implementation some of the initial `frontend` service APIs
- Since the frontend shares common objects that needs to be communicated with the **payments** and the **inventory** service, move the common classes into an SDK as a separate project
- Update skaffold configurations

### Architecture
image

***Notes on the Architecture:***
- The initial MVP will implement the data-store in memory. We will work on the expanding the app with an external storage option after the MVP
- The app exposes a _/switch_ functionality to switch between store types
- More details at: [go/abm-edge-app](http://go/abm-edge-app)

### CODE REVIEW

File: point-of-sale-app/frontend/src/main/java/com/google/abmedge/frontend/FrontendController.java [hunk 1]
```diff
@@ -14,19 +14,138 @@
...
+              switchEndpoint), e);
+    }
+    return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+  }
+
+  @PostMapping(value = "/pay")
+  public ResponseEntity<Void> pay(@RequestBody List<PurchaseItem> purchaseList) {
+    String updateEndpoint = INVENTORY_SERVICE + UPDATE_EP;
+    String payEndpoint = PAYMENTS_SERVICE + PAY_EP;
+    // try {
```
Comment (bourgeoisor) [inline point-of-sale-app/frontend/src/main/java/com/google/abmedge/frontend/FrontendController.java:124]: Is this placeholder taken care of in a following PR?

**Review by bourgeoisor — DISMISSED** [review]
Looks good, my big comment is to add class and method comments!

File: point-of-sale-app/frontend/src/main/java/com/google/abmedge/frontend/FrontendController.java (hunk 1 above)
Comment (Shabirmean) [reply point-of-sale-app/frontend/src/main/java/com/google/abmedge/frontend/FrontendController.java]: Yes this was implemented in the followup PRs. I have removed it in this PR completely!

**Review by puneith — APPROVED** [review]
lgtm

**Review by bourgeoisor — APPROVED** [review]
LGTM

**Pull request merged**